rclpy/src/rclpy/_rclpy.c

```diff
@@ -1345,6 +1345,20 @@ rclpy_destroy_node_entity(PyObject * Py_UNUSED(self), PyObject * args)
   }
 
   rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pynode, NULL);
+
+  if (!node) {
+    PyErr_SetString(PyExc_ValueError,
+      "rcl node is null");
+    return NULL;
+  }
+
+  if (!rcl_node_is_valid(node)) {
+    PyErr_SetString(PyExc_ValueError,
+      "rcl node is invalid");
+    rcl_reset_error();
+    return NULL;
+  }
```

Comment:
I think both of these conditionals are redundant, as the underlying fini function for the entity should check these things. It should already be reported up to Python by the `Failed to fini '%s': %s` error below, I think.